src/hooks/useFetch.js

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+
+const fetchUrl = (url, after, count, limit = 100, time = 'year') => {
+  const urlParams = `t=${time}&limit=${limit}${
+    count && after ? `&after=${after}&count${count}` : ''
+  }`;
+  return `${url}/top.json?${urlParams}`;
+};
+
+const makeRequest = async (url) => {
+  const res = await fetch(url);
+
+  if (!res.ok) {
+    throw new Error('Unable to fetch data');
+  }
+
+  const data = await res.json();
+  return data;
+};
+
+const fetchPosts = async (
+  url,
+  posts = [],
+  count = null,
+  after = null,
+  numRequests = 500,
```

Comment:
**Naming:** This reads like the number of requests. But there is only a max number of requests of 5